pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -59,8 +59,10 @@ func SetClusterConfigDefaults(cfg *ClusterConfig) {
 		cfg.VPC.ManageSharedNodeSecurityGroupRules = Enabled()
 	}
 
-	if cfg.Karpenter != nil && cfg.Karpenter.CreateServiceAccount == nil {
-		cfg.Karpenter.CreateServiceAccount = Disabled()
+	if cfg.Karpenter != nil {
```

Comment:
What prompted this change?

Reply:
I didn't completely revert the change there appropriately. I had originally added additional validation logic there for the initial boolean flag. I've changed this back since I assume there is no defaulting of the value since its now a string.